### PR TITLE
Don't use temporary directory for $HOME

### LINKED ISSUE #181: $HOME no longer works at all in config.yaml
Part of preventing writing to the user's home directory was to make k2 use a tempdir for $HOME for applications that don't allow configuration changes by any means.

This breaks config.yaml references that use $HOME.

### PR DESCRIPTION
**Pull request by joejulian**
Revert `mkdir -d` substitution for $HOME

Document how to map the needed files instead of $HOME
    
Add documentation to the README.md for binding just the files we need
for K2 instead of the entire home directory. This prevents the K2
processes from writing potentially overwriting files in the user's home
directory when running as a container.
    
Fixes #181 and is associated with #97
    
Similar changes to k2cli should be done, https://github.com/samsung-cnct/k2cli/issues/51

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 [![Docker Repository on Quay](https://quay.io/repository/samsung_cnct/k2/status "Docker Repository on Quay")](https://quay.io/repository/samsung_cnct/k2)
 
+You should not run K2 directly. Use [k2cli](https://github.com/samsung-cnct/k2cli). The following
```
Comment (coffeepac) [inline README.md]: The first section of the 'Getting Started' section directs users to k2cli.

@iredwards do you have an opinion on this?  Should there be two sections pointing people to k2cli and should this be gentler by removing the first section?

Comment (joejulian) [inline README.md]: I believe we should not waste the reader's time. Let them know right up front that they should be using k2cli.

Comment (coffeepac) [inline README.md]: I'm balking at the line starting with 'You should not run K2'.  I know the sentence carries on but that's jarring.

'You should use k2cli instead of K2 directly.  The following...'  That okay?

Comment (joejulian) [inline README.md]: Absolutely. Positive statements are always better.

**Comment by joejulian** [comment]
Go go Jenkins!

**Comment by coffeepac** [comment]
retest this please

**Comment by joejulian** [comment]
Closing for now since we do not yet have a `do not merge` label and I've found another problem with this that needs addressed first.

**Comment by joejulian** [comment]
Go go Jenkins!

**Comment by coffeepac** [comment]
retest this please

**Comment by coffeepac** [comment]
@joejulian does 'Go go Jenkins!' restart the testing process?

**Comment by joejulian** [comment]
Yes it does.

**Comment by joejulian** [comment]
Go go Jenkins!

**Comment by joejulian** [comment]
Go go Jenkins!

**Comment by joejulian** [comment]
I should change the Jenkins keyword to "banana" in hopes that if you give the test monkey enough bananas he'll eventually let your PR pass.

**Comment by davidewatson** [comment]
Please retest this

**Comment by davidewatson** [comment]
Please retest this

**Comment by davidewatson** [comment]
retest this please

**Comment by davidewatson** [comment]
CI testing are still failing due to terraform being unable to teardown the cluster: ``` 20:16:35 FAILED - RETRYING: TASK: /kraken/ansible/roles/kraken.provider/kraken.provider.aws : Run terraform destroy (10 retries left).
 ```

**Comment by davidewatson** [comment]
retest this please

**Comment by davidewatson** [comment]
retest this please

File: README.md [hunk 2]
```diff
@@ -73,13 +73,30 @@ The easiest way to get started with K2 directly is to use a K2 container image
...
+        -it"
+
 ### Initial K2 Directory
 If this is your first time using K2, use the K2 Docker image to generate a 'sensible defaults' configuration (this assumes AWS is the infrastructure provider):
 
 With the Docker container:
 
 ```bash
-docker run -v ~:/root --rm=true -it quay.io/samsung_cnct/k2:latest ./up.sh --generate
+docker run $K2OPTS quay.io/samsung_cnct/k2:latest ./up.sh --generate
```
Comment (davidewatson) [inline README.md]: If these instructions are really for development, then maybe we should document how to run your own code, instead of whatever happens to be in quay.io/samsung_cnct/k2:latest at the moment. For example ```docker run ${K2OPTS} -v /Users/dwat/workspace/k2:/workspace --rm -it quay.io/samsung_cnct/k2-tools bash -c "cd /workspace/ && ./up.sh" 
```

We can leave that for a separate PR for now though.

**Comment by davidewatson** [comment]
LGTM if it passes tests

**Comment by joejulian** [comment]
retest this please

**Comment by joejulian** [comment]
Go go Jenkins!

**Comment by coffeepac** [comment]
LGTM
